packages/store/src/internal/state-context-factory.ts

```diff
@@ -27,6 +27,18 @@ export class StateContextFactory {
 
     function setStateValue(currentAppState: any, newValue: T): any {
       const newAppState = setValue(currentAppState, metadata.depth, newValue);
+
+      if (metadata.instance && metadata.instance.ngxsOnChanges) {
+        const instance: NgxsLifeCycle = metadata.instance;
```

Comment:
If you move the `instance` variable outside the `if` block then it becomes simpler and you won't need the `!` later on the invocation of the `instance.ngxsOnChanges(change)` because TypeScript can now infer it from the check in the condition.
```TS
      const instance: NgxsLifeCycle = metadata.instance;
      if (instance && instance.ngxsOnChanges) {        
```

Reply:
@splincode Are you going to do this tweak?